homeassistant/components/mqtt/sensor.py

```diff
@@ -348,8 +373,9 @@ def native_unit_of_measurement(self) -> str | None:
         return self._config.get(CONF_UNIT_OF_MEASUREMENT)
 
     @property
-    def native_value(self) -> StateType | datetime:
+    def native_value(self) -> StateType | date | datetime:
         """Return the state of the entity."""
+        assert not isinstance(self._state, Decimal)
```

Comment:
Why add an assert here? It would make more sense to add Decimal to the possible return types.